Add more ways to detect runtime directory

wojciechkepka opened the pull request:
Closes: #212

archseer commented on helix-core/src/lib.rs:
```diff
@@ -48,15 +51,26 @@ pub fn find_root(root: Option<&str>) -> Option<std::path::PathBuf> {
 
 #[cfg(not(embed_runtime))]
 pub fn runtime_dir() -> std::path::PathBuf {
```
Can you make this private now? We always want to use the cached variable

wojciechkepka replied:
Done.

archseer commented on helix-core/src/lib.rs:
```diff
@@ -48,15 +51,26 @@ pub fn find_root(root: Option<&str>) -> Option<std::path::PathBuf> {
+        return conf_dir;
+    }
+
+    if let Ok(dir) = std::env::var("CARGO_MANIFEST_DIR") {
```
Nice! This is only via 'cargo' right?

wojciechkepka replied:
Yes, it’s exported only during cargo build/run.

pickfire replied:
Good idea. I didn't thought about this, if we run from cargo we don't need to use the runtime dir with this.

pickfire replied:
Good idea. I didn't thought about this, if we run from cargo we don't need to use the runtime dir with this.

archseer commented:
Sorry, can you also update the install part of the readme? The text regarding needing to set the var in development is no longer relevant, and for regular installs we can point out to the user that it's possible to either copy or symlink the `runtime/` dir under the config dir.

wojciechkepka commented:
Good idea, updated the README. I also changed how the directory is picked up when using executable dir.  From README I understood that you can have this structure:
```
bin
└── hx
└── runtime
```

And in reality it expected the runtime directory to be "extracted" alongside of hx. So I changed it to use `runtime` directory instead. Should I revert it to use the parent directory?

pickfire commented on README.md:
```diff
@@ -41,14 +41,12 @@ cargo install --path helix-term
-> running via cargo).
-> 
-> `export HELIX_RUNTIME=$PWD/runtime`
+Now copy the `runtime/` directory somewhere. Helix will by default look for the runtime
+inside the config directory or the same directory as executable, but that can be overriden
+via the `HELIX_RUNTIME` environment variable.
```
Maybe you might want to mention that one can just soft-link the runtime to the git repo runtime to be easier?

pickfire commented:
Looks good to me. We could suggest the soft link thing but not necessary for this patch.

archseer commented:
> And in reality it expected the runtime directory to be "extracted" alongside of hx. So I changed it to use `runtime` directory instead. Should I revert it to use the parent directory?

Good catch! ...this was certainly a bug and explains why there was issues with using binary releases directly from Github.
